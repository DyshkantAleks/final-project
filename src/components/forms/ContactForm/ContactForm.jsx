import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { Form, Input, Select, Row, Col, Radio } from 'antd';
import { Button } from '../../Button';
import styled from 'styled-components';
import { selectCustomer, selectCustomerIslogined } from '../../../store/customer/slectors';



export const ContactForm = (props) => {
  const { handleSubmit } = props;
  const logined = useSelector(selectCustomerIslogined)
  const customer = useSelector(selectCustomer)
  const [form] = Form.useForm();
  const [delivery, setdelivery] = useState('Self')
  const prefixSelector = (
    <Form.Item name='prefix' noStyle>
      <Select
        showArrow={false}
        style={{
          width: 76,
        }}
      />
    </Form.Item>
  );
  const deliveryMethod = [
    {
      text: 'Самовывоз',
      value: 'Self',
    },
    {
      text: 'Новая почта',
      value: 'Nova Poshta',
    },
    {
      text: 'УкрПочта',
      value: 'ukrPoshta',
    },
    {
      text: 'Курьероом на дом',
      value: 'toHome',
    },
  ];
  
  const payMethod = [
    {
      text: 'Наличными при получении',
      value: 'cash',
    },
    {
      text: 'Кредитная карта',
      value: 'creditCard',
    },
  ];
  const {name, surname, email, phone} = customer
  const initialValues = logined ? {name, surname, email, phone, prefix: '+380', delivery: 'Self' } : {prefix: '+380', delivery: 'Self'}
  console.log(delivery)
  return (
    <Form
      layout='vertical'
      name='order'
      form={form}
      onFinish={handleSubmit}
      onFinishFailed={val =>({...initialValues, ...val}) }
      initialValues={initialValues}
    >
      <Text>Пожалуйста, заполните форму</Text>
      <Text>Выберите форму доставки и оплаты</Text>

      <Row gutter={10}>
        <Col span={12}>
          <Form.Item
            name='name'
            label='Имя'
            rules={[
              {
                required: true,
                message: 'Введите имя!',
              },
              {
                pattern: /^[a-zA-Zа-яёА-ЯЁІіїЇєЄ]+$/,
                message: 'Имя должно состоять из букв a-z, A-Z, а-я, А-Я!',
              },
              {
                min: 2,
                message: 'Имя должно содержать минимум 2 символа!',
              },
              {
                max: 25,
                message: 'Имя должно содержать максимум 25 символов!',
              }
            ]}
          >
            <Input />
          </Form.Item>
        </Col>
        <Col span={12}>
          <Form.Item
            name='surname'
            label='Фамилия'
            rules={[
              {
                required: true,
                message: 'Введите фамилию!',
              },
              {
                pattern: /^[a-zA-Zа-яёА-ЯЁІіїЇєЄ]+$/,
                message: 'Фамилия должна состоять из букв a-z, A-Z, а-я, А-Я!',
              },
              {
                min: 2,
                message: 'Фамилия должна содержать минимум 2 символа!',
              },
              {
                max: 25,
                message: 'Фамилия должна содержать максимум 25 символов!',
              }
            ]}
          >
            <Input />
          </Form.Item>
        </Col>
      </Row>

      <Row gutter={10}>
        <Col span={12}>
          <Form.Item
            name='email'
            label='E-mail'
            rules={[
              {
                type: 'email',
                message: 'Некоректный E-mail ',
              },
              {
                required: true,
                message: 'Введите  E-mail!',
              },
            ]}
          >
            <Input />
          </Form.Item>
        </Col >
        <Col span={12}>
          <Form.Item
            name='phone'
            label='Телефон'
            rules={[{ required: true, message: 'Введите номер телефона!' }]}
          >
            <Input addonBefore={prefixSelector} />
          </Form.Item>
        </Col>
      </Row>
      <Row gutter={10}>
        {(!(delivery === 'Self')) && <Col span={12}>
          <Form.Item
            name='city'
            label='Населенный пункт'
            rules={[
              {
                required: true,
                message: 'Заполните поле!',
              },
              {
                pattern: /^[а-яёА-ЯЁІіїЇєЄ]+$/,
                message: 'заполните поле кирилицей',
              },
            ]}
          >
            <Input />
          </Form.Item>
        </Col>}
        {(!(delivery === 'Self')) && <Col span={12}>
          <Form.Item
            name='street'
            label='Улица'
            rules={[
              {
                required: true,
                message: 'Заполните поле!',
              },
              {
                pattern: /^[а-яёА-ЯЁІіїЇєЄ]+$/,
                message: 'Заполните поле кирилицей',
              },
            ]}
          >
            <Input />
          </Form.Item>
        </Col>}
      </Row>
      <Row gutter={10}>
        {(!(delivery === 'Self')) && <Col span={12}>
          <Form.Item
            name='house'
            label='Дом'
            rules={[
              {
                required: true,
                message: 'Заполните поле!',
              },
              {
                pattern: /^[0-9а-яёА-ЯЁІіїЇєЄ]+$/,
                message: 'Используйте цифры и буквы',
              },
            ]}
          >
            <Input />
          </Form.Item>
        </Col>}
        {(!(delivery === 'Self')) && <Col span={12}>
          <Form.Item
            name='flat'
            label='Квартира'
            rules={[
              {
                
                message: 'Заполните поле!',
              },
              {
                pattern: /^[0-9]+$/,
                message: 'Используйте цифры',
              }
            ]}
          >
            <Input />
          </Form.Item>
        </Col>}
      </Row>
      <Row>
        <Col span={12}>
          <Form.Item
            name='delivery'
            label='Способ доставки'
            rules={[{ message: 'Выбирите способ доставки!' }]}
          >
            <Radio.Group onChange={(val) => setdelivery(val.target.value)}>
              {deliveryMethod.map(({text, value}, index) => (<Radio style={radioStyle} key={index} value={value}>{text}</Radio>))}
            </Radio.Group>
          </Form.Item>
        </Col>
        <Col span={12}>
          <Form.Item
            name='payMethod'
            label='Способ оплаты'
            rules={[{ message: 'Выбирите способ оплаты!' }]}
          >
            <Radio.Group>
              {payMethod.map(({text, value}, index) => (<Radio style={radioStyle} key={index} value={value}>{text}</Radio>))}
            </Radio.Group>
          </Form.Item>
        </Col>
      </Row>
      
      <Button text='Подтвердить заказ' type='submit'></Button>
      
    </Form>
    
  );
};

const Text = styled.p`
  font-size: 1.4rem;
  line-height: 1.5;
`;
const radioStyle = {
  display: 'block',
  height: '30px',
  lineHeight: '30px',
};
