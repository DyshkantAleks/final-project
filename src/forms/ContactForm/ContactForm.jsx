import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { Form, Input, Select, Row, Col, Radio, AutoComplete } from 'antd';
import { Button } from '../../components/Button';
import styled from 'styled-components';
import { selectCustomer } from '../../store/customer/slectors';
import { GlobalConfig } from '../../GlobalConfig';
import { getCity, getStreet } from '../../utils/novaPoshtaApi';
import { ValodationRules } from '../../validation/ValidationRules';

export const ContactForm = (props) => {
  const { handleSubmit } = props;
  const customer = useSelector(selectCustomer);
  const { name, surname, email, phone } = customer;

  const searchCode = GlobalConfig.deliveryOptions[2].NOVA_POSHTA.serchCityCode;
  const [autoCompleteCityResult, setAutoCompleteCityResult] = useState([]);
  const [autoCompleteStreetResult, setAutoCompleteStreetResult] = useState([]);
  const [delivery, setdelivery] = useState(
    GlobalConfig.deliveryOptions[0].PICKUP.value
  );
  
  const [form] = Form.useForm();
  // autocomplete city
  const onCityChange = (value) => {
    if (!GlobalConfig.adressFieldRegExp.test(value)) {
      setAutoCompleteCityResult([]);
    } else {
      getCity(value).then((list) => {
        
        return setAutoCompleteCityResult(list.map((list) => list));
      });
    }
  };
  const cityOptions = autoCompleteCityResult.map((city) => ({
    label: city,
    value: city,
  }));
  // ------------------------------------------
  // autocomplete street
  const onStreetChange = (value) => {
    if (!GlobalConfig.adressFieldRegExp.test(value)) {
      setAutoCompleteStreetResult([]);
    } else {
      getStreet(value, searchCode).then((list) => {
      
        setAutoCompleteStreetResult(
          list.map((list) => {
            return list;
          })
        );
      });
    }
  };
  const streetOptions = autoCompleteStreetResult.map((street) => ({
    label: street,
    value: street,
  }));
  // --------------------------------------------
  const isVisibleAdressField =
  delivery === GlobalConfig.deliveryOptions[0].PICKUP.value;
  
  const prefixSelector = (
    <Form.Item name='prefix' noStyle>
      <Select
        showArrow={false}
        style={{
          width: 76,
        }}
      />
    </Form.Item>
  );
  return (
    <Form
      layout='vertical'
      name='order'
      form={form}
      onFinish={handleSubmit}
      initialValues={{
        name,
        surname,
        email,
        phone,
        prefix: '+380',
        delivery: GlobalConfig.deliveryOptions[0].PICKUP.value,
        payMethod: GlobalConfig.paymentOptions[0].BY_CASH.value,
      }}
    >
      <Text>Пожалуйста, заполните форму</Text>
      <Text>Выберите форму доставки и оплаты</Text>

      <Row gutter={10}>
        <Col span={12}>
          <Form.Item
            name='name'
            label='Имя'
            rules={ValodationRules.nameRules}
          >
            <Input />
          </Form.Item>
        </Col>
        <Col span={12}>
          <Form.Item
            name='surname'
            label='Фамилия'
            rules={ValodationRules.surnameRules}
          >
            <Input />
          </Form.Item>
        </Col>
      </Row>

      <Row gutter={10}>
        <Col span={12}>
          <Form.Item
            name='email'
            label='E-mail'
            rules={ValodationRules.emailRules}
          >
            <Input />
          </Form.Item>
        </Col>
        <Col span={12}>
          <Form.Item
            name='phone'
            label='Телефон'
            rules={ValodationRules.phoneRules}
          >
            <Input addonBefore={prefixSelector} />
          </Form.Item>
        </Col>
      </Row>
      {!isVisibleAdressField && (
        <>
          <Row gutter={10}>
            <Col span={12}>
              <Form.Item
                name='city'
                label='Населенный пункт'
                rules={[
                  {
                    required: isVisibleAdressField,
                    message: 'Заполните поле!',
                  },
                  {
                    pattern: GlobalConfig.adressFieldRegExp,
                    message: 'заполните поле кирилицей',
                  },
                ]}
              >
                <AutoComplete options={cityOptions} onChange={onCityChange}>
                  <Input onChange={onCityChange} />
                </AutoComplete>
              </Form.Item>
            </Col>

            <Col span={12}>
              <Form.Item
                name='street'
                label='Улица'
                rules={[
                  {
                    required: isVisibleAdressField,
                    message: 'Заполните поле!',
                  },
                  {
                    pattern: GlobalConfig.adressFieldRegExp,
                    message: 'Заполните поле кирилицей',
                  },
                ]}
              >
                <AutoComplete options={streetOptions} onChange={onStreetChange}>
                  <Input />
                </AutoComplete>
              </Form.Item>
            </Col>
          </Row>
          <Row gutter={10}>
            <Col span={12}>
              <Form.Item
                name='house'
                label='Дом'
                rules={[
                  {
                    required: isVisibleAdressField,
                    message: 'Заполните поле!',
                  },
                  {
                    pattern: GlobalConfig.adressFieldRegExp,
                    message: 'Используйте цифры и буквы',
                  },
                ]}
              >
                <Input />
              </Form.Item>
            </Col>

            <Col span={12}>
              <Form.Item
                name='flat'
                label='Квартира'
                rules={[
                  {
                    message: 'Заполните поле!',
                  },
                  {
                    pattern: GlobalConfig.numberFieldRegExp,
                    message: 'Используйте цифры',
                  },
                ]}
              >
                <Input />
              </Form.Item>
            </Col>
          </Row>
        </>
      )}
      <Row>
        <Col span={12}>
          <Form.Item
            name='delivery'
            label='Способ доставки'
            rules={[{ message: 'Выбирите способ доставки!' }]}
          >
            <Radio.Group onChange={(val) => setdelivery(val.target.value)}>
              {GlobalConfig.deliveryOptions
                .reduce((acc, rec, index) => {
                  return acc.concat(Object.values(rec));
                }, [])
                .map(({ text, value }, index) => (
                  <Radio style={radioStyle} key={index} value={value}>
                    {text}
                  </Radio>
                ))}
            </Radio.Group>
          </Form.Item>
        </Col>
        <Col span={12}>
          <Form.Item
            name='payMethod'
            label='Способ оплаты'
            rules={[{ message: 'Выбирите способ оплаты!' }]}
          >
            <Radio.Group>
              {GlobalConfig.paymentOptions
                .reduce((acc, rec, index) => {
                  return acc.concat(Object.values(rec));
                }, [])
                .map(({ text, value }, index) => (
                  <Radio style={radioStyle} key={index} value={value}>
                    {text}
                  </Radio>
                ))}
            </Radio.Group>
          </Form.Item>
        </Col>
      </Row>
      <Button text='Подтвердить заказ' type='submit'></Button>
    </Form>
  );
};
const Text = styled.p`
  font-size: 1.4rem;
  line-height: 1.5;
`;
const radioStyle = {
  display: 'block',
  height: '30px',
  lineHeight: '30px',
};
