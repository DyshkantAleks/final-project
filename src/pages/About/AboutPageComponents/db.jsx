export const qwerty = {
  text: '<h2>МЫ - КОМАНДА МОЛОДЫХ УВЛЕЧЕННЫХ ПРОФЕССИОНАЛОВ</h2>',
};
