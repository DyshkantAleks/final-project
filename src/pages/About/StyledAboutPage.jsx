import styled from "styled-components";
import {device} from "../../styles/breakpoints/breakpoints";

export const ContactLink = styled.a`
font-size: 1.4rem;
line-height: 1.5;
text-decoration: underline;
color: inherit;
`;

export const TextContainer = styled.div`
flex-basis: 60%;
margin-right: 5rem;
`;
export const ContainerPage = styled.div`
display: flex;
flex-direction: column;
@media${device.tabletM}{
justify-content: space-between;
flex-direction: row;
}
`;
export const ContainerContactForm = styled.div`
border: .1rem solid #aecfd9;
box-shadow: 5px 5px 10px #aecfd9 ;
padding: 2rem;
@media${device.tabletM}{
width: 30rem;
margin: 0 auto;
align-self: flex-start;
}
@media${device.tabletM}{
margin: 0;
}
`;
export const ContactForm = styled.p`
font-size: 1.4rem;
line-height: 1.5;
font-weight: 500;
@media${device.tabletM}{

}
`;


export const Text = styled.p`
font-size: 1.4rem;
line-height: 1.5;
`

export const TextList = styled.ul`
font-size: 1.4rem;
line-height: 1.5;


`
export const TextListItem = styled.li`

`