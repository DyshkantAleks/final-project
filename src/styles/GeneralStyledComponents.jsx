import styled from 'styled-components';
import { device } from './breakpoints/breakpoints';

export const ContentContainer = styled.div`
    padding: 0 1.6rem;
    margin: 0 auto;
    
    @media ${device.tabletM}{
        padding: 0 2.4rem;
    }
    @media ${device.tabletL}{
        padding: 0 5rem;
    }
    
    @media ${device.desktop}{
        padding: 4rem 0;
        width: 120rem;
    }
`