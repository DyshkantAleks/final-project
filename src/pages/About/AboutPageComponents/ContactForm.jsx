/* eslint-disable jsx-a11y/anchor-has-content */
import React from 'react';

import { ContactForm, ContactLink, ContainerContactForm } from '../StyledAboutPage';

export const ContactsForm = () => {
  return (
    <ContainerContactForm>
      <ContactForm>Хочешь узнать подробности? Связись с нами по телефону:</ContactForm>
      <ContactLink href='tel:+380442902244'>+38 (044) 290 22 44</ContactLink>
      <ContactForm>Или напишите нам на электронную почту: </ContactForm>
      <ContactLink href='mailto:office@dan-it.com.ua'>office@dan-it.com.ua</ContactLink>
      <ContactForm>Прихотите в наш салон по адресу: </ContactForm>
      <ContactLink
        href="https://www.google.com/maps/place/50%C2%B025'44.4%22N+30%C2%B035'37.1%22E/@50.4327083,30.5527535,13z"
        target='_blank' rel='noopener'
      >г. Киев, пр-т Павла Тычины, 1в, ТОЦ «Silver Breeze», офис А, 6-й этаж
      </ContactLink>
      <div>
        <a
          target='_blank'
          rel='noopener noreferrer'
          href='https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d2541.651640662533!2d30.59143871518244!3d50.42896179679486!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x40d4c57deccaada1%3A0xa5e7cf7c9f8ff18b!2sDAN.IT!5e0!3m2!1sru!2sua!4v1603998516813!5m2!1sru!2sua'
        />
      </div>
    </ContainerContactForm>
  )
};