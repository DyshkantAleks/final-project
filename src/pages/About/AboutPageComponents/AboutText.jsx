import React from 'react';

import { ContentContainer } from '../../../styles/GeneralStyledComponents';
import { ContainerPage, TextContainer, Text, TextList, TextListItem } from '../StyledAboutPage';
import { ContactsForm } from './ContactForm';

export const AboutText = () => {
  return (
    <ContentContainer>
      <ContainerPage>
        <TextContainer>
          <h2>МЫ - КОМАНДА МОЛОДЫХ УВЛЕЧЕННЫХ ПРОФЕССИОНАЛОВ</h2>
          <Text>Мы уверены, что пространство, в котором мы живем, должно отражать наши мечты и стремления, и каждый из
            нас
            в силах создать достойный восхищения интерьер, в котором приятно находиться и в который всегда хочется
            возвращаться.
          </Text>
          <Text>Наш опыт, стремление к комплексному подходу во всем, уважение к желаниям клиента позволяют нам
            помогать
            Вам реализовать любые ваши идеи в реальные сроки за разумные деньги.

          </Text>
          <Text>Мы работаем напрямую с лучшими фабриками Европы и Азии, производящими мебель и предметы интерьера
            ценового
            сегмента «средний+», «высокий», «премиум», что позволяет нам предлагать Вам актуальные новинки по
            конкурентным
            ценам. Большой выбор товаров в наличии и под заказ позволяют нам помочь Вам подобрать достойные варианты в
            рамках Вашего бюджета.
          </Text>
          <Text>Мы уже реализовали ряд проектов, основываясь на принципах:</Text>
          <TextList>
            <TextListItem>Комплексный подход эффективен всегда.</TextListItem>
            <TextListItem>Нереализуемых задач не бывает. Мы всегда готовы предложить варианты решения даже самой
              нетрадиционной задачи.
            </TextListItem>
            <TextListItem>Важен баланс качества результата, сроков реализации проекта и бюджета.</TextListItem>
          </TextList>
        </TextContainer>
        <ContactsForm />
      </ContainerPage>
    </ContentContainer>
  )
};