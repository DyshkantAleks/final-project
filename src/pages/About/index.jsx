import React from 'react';
import {Header} from '../../commons/Header/Header';
import {AboutText} from "./AboutPageComponents/AboutText";
import { ContentContairer } from '../../components/Content/Content';

export const AboutPage = () => {
  return (
    <ContentContairer>
      <Header/>
      <h1>About page</h1>
      <AboutText/>
    </ContentContairer>
  )
}