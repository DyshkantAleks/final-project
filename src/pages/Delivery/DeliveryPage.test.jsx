import React from 'react';
import { shallow } from 'enzyme';

import { DeliveryPage } from './index';
import { ContentContainer } from '../../styles/GeneralStyledComponents';
import { ContainerPage, Text, TextContainer, TextList, TextListItem } from '../About/StyledAboutPage';
import { ContactsForm } from '../About/AboutPageComponents/ContactForm';

const setUp = (props = {}) => shallow(<DeliveryPage {...props} />);

describe('DeliveryPage component', () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });
  it('should be render with general wrapper', () => {
    const generalWrapper = component.find(ContentContainer);
    expect(generalWrapper.length).toEqual(1);
  });
  it('should be render with wrapper', () => {
    const wrapper = component.find(ContainerPage);
    expect(wrapper.length).toEqual(1);
  });
  it('should be render with text container', () => {
    const textContainer = component.find(TextContainer);
    expect(textContainer.length).toEqual(1);
  });
  it('should be render with <h2>', () => {
    const title = component.find('h2');
    expect(title.length).toEqual(1);
    expect(title.text()).toEqual('ОБЩИЕ УСЛОВИЯ');
  });
  it('should be render with text', () => {
    const text = component.find(TextContainer).find(Text);
    expect(text.length).toEqual(4);
    expect(text.at(0).text()).toEqual('Мы доставляем заказы во все регионы Украины, Белоруссию, Россию. Кроме этого мы доставляем товар в Европу, США.');
    expect(text.at(1).text()).toEqual('Бесплатная доставка собственной курьерской службой для всех заказов стоимостью более 20000 руб. по Киеву в течении 2-5 дней.');
    expect(text.at(2).text()).toEqual('Бесплатная доставка собственной курьерской службой для всех заказов стоимостью более 15000 гривен. по Киевской области в течение 2-5 дней.');
    expect(text.at(3).text()).toEqual('КУРЬЕРСКАЯ ДОСТАВКА');
  });
  it('should be render with <ul>', () => {
    const textList = component.find(TextList);
    expect(textList.length).toEqual(2);
  });
  it('should render with list', () => {
    const textListItem = component.find(TextListItem);
    expect(textListItem.length).toEqual(8);
    expect(textListItem.at(0).text()).toEqual('Стоимость доставки складывается из курьерской доставки заказа и подъема на этаж.');
    expect(textListItem.at(1).text()).toEqual('Доставка крупных заказов (несколько позиций мебели) может осуществляться партиями, в зависимости от сроков готовности товара к отгрузке.');
    expect(textListItem.at(2).text()).toEqual('Занести мебель в квартиру (офис) можно только, если ширина дверного проема составляет не менее 75 см, а ширина коридора и других помещений - не менее 120 см. Несоответствие ширины дверных проемов, коридоров и других помещений указанным размерам не является основанием для отказа от товара. Съем дверей и перестановка предметов при заносе мебели оплачивается отдельно.');
    expect(textListItem.at(3).text()).toEqual('Если вы решили отказаться от товара, вы можете вернуть его в течение семи дней, однако вам нужно будет организовать доставку до нашего склада в Москве или оплатить обратную доставку.');
    expect(textListItem.at(4).text()).toEqual('В течение 2-5 рабочих дней после подтверждения заказа.');
    expect(textListItem.at(5).text()).toEqual('Оплата наличными или картой курьеру, безналичная оплата на сайте при заказе.');
    expect(textListItem.at(6).text()).toEqual('С 10:00 до 21:00 ежедневно (укажите удобное время в комментариях к заказу).');
    expect(textListItem.at(7).text()).toEqual('Доступные интервалы: с 12:00 до 17:00 или с 17:00 до 21:00.');
  });
  it('should render with form of contacts', () => {
    const contacts = component.find(ContactsForm);
    expect(contacts.length).toEqual(1);
  })
  it('snapshot <DeliveryPage />', () => {
    const generalWrapper = component.find(ContentContainer);
    expect(generalWrapper.debug()).toMatchSnapshot();
  });
});