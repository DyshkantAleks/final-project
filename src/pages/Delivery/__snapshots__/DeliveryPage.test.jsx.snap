// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryPage component snapshot <DeliveryPage /> 1`] = `
"<styled.div>
  <ScrollOnTop />
  <styled.div>
    <styled.div>
      <h2>
        ОБЩИЕ УСЛОВИЯ
      </h2>
      <styled.p>
        Мы доставляем заказы во все регионы Украины, Белоруссию, Россию. Кроме этого мы доставляем товар в Европу, США.
      </styled.p>
      <styled.ul>
        <styled.li>
          Стоимость доставки складывается из курьерской доставки заказа и подъема на этаж.
        </styled.li>
        <styled.li>
          Доставка крупных заказов (несколько позиций мебели) может осуществляться партиями, в зависимости от сроков готовности товара к отгрузке.
        </styled.li>
        <styled.li>
          Занести мебель в квартиру (офис) можно только, если ширина дверного проема составляет не менее 75 см, а ширина коридора и других помещений - не менее 120 см. Несоответствие ширины дверных проемов, коридоров и других помещений указанным размерам не является основанием для отказа от товара. Съем дверей и перестановка предметов при заносе мебели оплачивается отдельно.
        </styled.li>
        <styled.li>
          Если вы решили отказаться от товара, вы можете вернуть его в течение семи дней, однако вам нужно будет организовать доставку до нашего склада в Москве или оплатить обратную доставку.
        </styled.li>
      </styled.ul>
      <styled.p>
        Бесплатная доставка собственной курьерской службой для всех заказов стоимостью более 20000 руб. по Киеву в течении 2-5 дней.
      </styled.p>
      <styled.p>
        Бесплатная доставка собственной курьерской службой для всех заказов стоимостью более 15000 гривен. по Киевской области в течение 2-5 дней.
      </styled.p>
      <styled.p>
        КУРЬЕРСКАЯ ДОСТАВКА
      </styled.p>
      <styled.ul>
        <styled.li>
          В течение 2-5 рабочих дней после подтверждения заказа.
        </styled.li>
        <styled.li>
          Оплата наличными или картой курьеру, безналичная оплата на сайте при заказе.
        </styled.li>
        <styled.li>
          С 10:00 до 21:00 ежедневно (укажите удобное время в комментариях к заказу).
        </styled.li>
        <styled.li>
          Доступные интервалы: с 12:00 до 17:00 или с 17:00 до 21:00.
        </styled.li>
      </styled.ul>
    </styled.div>
    <ContactsForm />
  </styled.div>
</styled.div>"
`;
